docs/adr/0005-rapid-releases-and-moving-towards-a-monorepo.md

```diff
@@ -0,0 +1,83 @@
+
+  * Good, because code will get merged and released more frequently.
+  * Good, because we don't need to spend time cutting releases.
+  * Good, because uplifts can be done much quicker.
+  * Bad, because we need implement new automation.
+  * Bad, because when breaking changes cause the automatic nightly merge to fail for an application.  This means a developer will get the relatively urgent task of resolving things.
+  * Good, because this will unblock future changes from being merged and released on that application.
+  * Good, because the developer who writes code with a breaking change is responsible for getting the PRs merged to resolve it (rather than the developer who cuts the release).
+  * Bad, because breaking changes require the corresponding PRs to be approved from all other teams before they can be merged into application-services.
+  * Good, because this is how things will work in a mono-repo world.  Adjusting our workflow now prepares us for the future and will give us insights that may help guide the monorepo migration.
+  * Good, because our code will exist alongside the application code like firefox-android and firefox-ios, which is how it will be in a monorepo world.
+  * Good, because our version numbers will align more closely with the rest of Mozilla.
+
+## Note on breaking changes
+
+Breaking changes to our consumer application code require extra developer work for both the current and new release models.  We should use this time to restructure our code to prevent breaking changes when it makes sense.  This means creating a more stable API boundary between us and our consumers and evolving those API boundaries in a backwards compatible way.  Some examples of this:
+
+  * When we did the recent exception class refactoring, we tried to be intentional about what exceptions we were exposing to consumers.  The goal was to only expose exceptions that were useful and that we expected to be able to support for a while.
+  * The exception reporting rework was done in backward compatible stages: first we added the new handler code and got the applications to start using it.  Then we get the applications to stop using the old code.  Finally, we remove the old code.  Nimbus has also been using techniques like this to evolve their API.
+  * In hindsight, our SQLCipher logins work could have been split into 2 parts.  The first part would have reworked the API of the logins wrapper code to something that we could have supported before the SQLCipher rework, after the rework, and also going forward into the future.  This would introduce breaking changes for our consumers and we could create PRs to discuss and resolve them.  While those PRs were in review, we could start on the actual SQLCipher changes.  Assuming the initial PRs were approved and merged before we were done with the second round of work, we could then merge that work to main as we pleased, without causing any breaking changes.
```

Comment:
This section almost wants to be a separate ADR or a document, I think there is a lot of value here about breaking changes and we don't want it overshadowed by the conversation on how we ship

Reply:
I was actually planning it to be a separate ADR, but I wasn't sure exactly how to structure that one since it's not so much a decision but more of a recommendation.  I'll give it a try though.

Reply:
I ended up just dropping this section because I wasn't sure what to do with it.  It doesn't feel like it belongs in an ADR or a HOWTO to me -- especially since I don't see much value in it.  It seems pretty unlikely that someone is going to read the doc before making a breaking change and decide to restructure their code in some way.